qcengine/programs/xtb.py

```diff
@@ -0,0 +1,55 @@
+"""
+Harness for the extended tight binding (xtb) package.
+This implementation interfaces with the xtb C-API via the xtb-python project,
+which is providing Python bindings and native extensions for common computational
+chemistry frameworks, like QCEngine.
+
+Therefore, this harness only has to provide a thin wrapper to integrate xtb.
+For more information on xtb-python and the actual QCSchema integration,
+visit `its documentation <https://xtb-python.readthedocs.io>`_.
+"""
+
+from qcelemental.models import AtomicInput, AtomicResult
+from qcelemental.util import which_import
+
+from .model import ProgramHarness
+from ..config import TaskConfig
+
+
+class XTBHarness(ProgramHarness):
+    """Calculation harness for the extended tight binding (xtb) package."""
+
+    _defaults = {
+        "name": "xtb",
+        "scratch": False,
+        "thread_safe": True,
+        "thread_parallel": False,
+        "node_parallel": False,
+        "managed_memory": False,
+    }
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    @staticmethod
+    def found(raise_error: bool = False) -> bool:
+        """Check for the availability of the Python API of xtb"""
+
+        return which_import(
+            "xtb",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="Please install via `conda install xtb-python -c conda-forge`.",
+        )
+
+    def compute(self, input_data: AtomicInput, config: TaskConfig) -> AtomicResult:
+        """
+        Actual interface to the xtb package. The compute function is just a thin
+        wrapper around the native QCSchema interface of xtb-python.
+        """
+
+        self.found(raise_error=True)
+        import xtb
+        from xtb.qcschema.harness import run_qcschema
+
+        return run_qcschema(input_data)
```

Comment:
Note that handling successful xtb runs is comparatively easy to managing error runs (either of execution or job). 

For orientation looking at psi models, the old way (which _doesn't_ depend on qcng in psi itself and only supports executable mode, `psi4 --json`) is handled https://github.com/MolSSI/QCEngine/blob/master/qcengine/programs/psi4.py#L133-L185 and https://github.com/psi4/psi4/blob/master/psi4/driver/schema_wrapper.py#L431-L479. 

Whereas the newer way (which _does_ depend on qcng `model_wrapper` and supports both executable mode and python mode, `psi4.run_qcschema`, which is more like what you want) is handled in https://github.com/MolSSI/QCEngine/blob/master/qcengine/programs/psi4.py#L189-L234 and https://github.com/psi4/psi4/blob/master/psi4/driver/schema_wrapper.py#L381-L428).

Reply:
I guess, most of this should get addressed on the `xtb-python` side. There is no command line interface for `xtb-python` yet and I don't really plan to add one in the near future, also making an IO harness for the real `xtb` binary is out-of-scope for this PR.

For `qcng` it would be advantageous to (optionally) wrap the API call in a subprocess in case there is an error, which is not caught in the API and returned properly as error in the `AtomicResult`. This is definitely something I have to look into.